### PR TITLE
Vg/ec 183

### PR DESCRIPTION
**Pull request by vinaygupta1504**
@yogendra689, @mauryaranjeet, @ggerig , @darshan-patel 
Kindly review the Bright Cove Content Integration.

### CODE REVIEW

**Comment by yogendra689** [comment]
looks good

File: app/service/bright_cove_integration.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  def content_item_attributes(entry)
+    deep_link_url = get_direct_link(entry['id'])
+    {
+      external_id: entry['id'],
+      name: sanitize_content(entry['name']),
+      description: sanitize_content(entry['description']),
+      summary:  sanitize_content(entry['long_description']),
+      url: deep_link_url,
+      # tags: entry['tags'].present? ? [{ 'source' => 'native', 'tag_type' => 'keyword', 'tag' => entry['tags'] }] : nil,
```
Comment (yogendra689) [inline app/service/bright_cove_integration.rb]: remove unwanted code

File: app/service/bright_cove_integration.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          url: deep_link_url,
+          title: sanitize_content(entry['name']),
+          description: sanitize_content(entry['description']) #,
+          # images: [{ url: entry['images']['poster']}, { url: entry['images']['thumbnail']}]
+      },
+
+      additional_metadata: {
+        # url: [{entry['published_at']}, {url: entry['state']}, {url: entry['complete']}]
+        created_at: entry['created_at'],
+        updated_at: entry['updated_at']
```
Comment (yogendra689) [inline app/service/bright_cove_integration.rb]: do consider to use updated_at  to compare with source last polled at 
push only delta changes

**Comment by mikkhait** [comment]
please follow proper naming convention for PRs

JIRA: JIRA-TITLE

File: app/service/bright_cove_integration.rb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    players['items'].first['url']
+  end
+
+  def get_content(options = {})
+    begin
+      params = { page: options[:page], limit: options[:limit], offset: options[:start] }
+      params['q'] = "updated_at:#{options[:last_polled_at]}..now" if @credentials['is_delta'].nil? || @credentials['is_delta'] == 'true'
+      videos = json_request(videos_url, :get, headers: { 'Authorization' => "Bearer #{get_access_token}" }, params: params)
+      if videos.count > 0
+        videos.map { |entry| create_content_item(entry) if Time.parse(options[:last_polled_at]) < Time.parse(entry['updated_at']) }
```
Comment (yogendra689) [inline app/service/bright_cove_integration.rb]: condiiton check should be in separate method to make it more readable

**Comment by yogendra689** [comment]
Please let us know if brightcove is content provider or LMS?

**Pull request merged**